### PR TITLE
Don't install tests subpackages either

### PR DESCRIPTION
**Pull request by thesamesam**
We already try to avoid installing tests but we need to include a glob to catch subpackages (we do
the same in neo4j's boltkit).

### CODE REVIEW

**Comment by robsdedude** [comment]
Hello and thanks for taking the time to open this PR. Before I can even review it, I must ask you to sign our [CLA (Contributor License Agreement)](https://neo4j.com/developer/cla/).

**Comment by thesamesam** [comment]
@robsdedude Done, thanks.

File: setup.py [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "console_scripts": [
     ],
 }
-packages = find_packages(exclude=["tests"])
+packages = find_packages(exclude=["tests", "tests.*"])
```
Comment (robsdedude) [inline setup.py]: So when I run `python setup.py sdist` with and without the change, I both times get 
```
$ python setup.py sdist
running sdist
running egg_info
writing neo4j.egg-info/PKG-INFO
writing dependency_links to neo4j.egg-info/dependency_links.txt
writing requirements to neo4j.egg-info/requires.txt
writing top-level names to neo4j.egg-info/top_level.txt
reading manifest file 'neo4j.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no previously-included files matching '*.class' found anywhere in distribution
warning: no previously-included files matching '*.pyc' found anywhere in distribution
warning: no previously-included files matching '*.pyo' found anywhere in distribution
warning: no previously-included files matching '*.so' found anywhere in distribution
warning: no previously-included files matching '*.dll' found anywhere in distribution
warning: no previously-included files matching '__pycache__' found anywhere in distribution
no previously-included directories found matching 'tests'
adding license file 'LICENSE.txt'
adding license file 'NOTICE.txt'
writing manifest file 'neo4j.egg-info/SOURCES.txt'
running check
creating neo4j-5.2.dev0
creating neo4j-5.2.dev0/neo4j
creating neo4j-5.2.dev0/neo4j.egg-info
creating neo4j-5.2.dev0/neo4j/_async
creating neo4j-5.2.dev0/neo4j/_async/io
creating neo4j-5.2.dev0/neo4j/_async/work
creating neo4j-5.2.dev0/neo4j/_async_compat
creating neo4j-5.2.dev0/neo4j/_async_compat/network
creating neo4j-5.2.dev0/neo4j/_async_compat/shims
creating neo4j-5.2.dev0/neo4j/_codec
creating neo4j-5.2.dev0/neo4j/_codec/hydration
creating neo4j-5.2.dev0/neo4j/_codec/hydration/_interface
creating neo4j-5.2.dev0/neo4j/_codec/hydration/v1
creating neo4j-5.2.dev0/neo4j/_codec/hydration/v2
creating neo4j-5.2.dev0/neo4j/_codec/packstream
creating neo4j-5.2.dev0/neo4j/_codec/packstream/v1
creating neo4j-5.2.dev0/neo4j/_spatial
creating neo4j-5.2.dev0/neo4j/_sync
creating neo4j-5.2.dev0/neo4j/_sync/io
creating neo4j-5.2.dev0/neo4j/_sync/work
creating neo4j-5.2.dev0/neo4j/graph
creating neo4j-5.2.dev0/neo4j/spatial
creating neo4j-5.2.dev0/neo4j/time
creating neo4j-5.2.dev0/neo4j/work
creating neo4j-5.2.dev0/testkitbackend
creating neo4j-5.2.dev0/testkitbackend/_async
creating neo4j-5.2.dev0/testkitbackend/_sync
copying files to neo4j-5.2.dev0...
copying LICENSE.txt -> neo4j-5.2.dev0
copying MANIFEST.in -> neo4j-5.2.dev0
copying NOTICE.txt -> neo4j-5.2.dev0
copying README.rst -> neo4j-5.2.dev0
copying setup.cfg -> neo4j-5.2.dev0
copying setup.py -> neo4j-5.2.dev0
copying neo4j/__init__.py -> neo4j-5.2.dev0/neo4j
copying neo4j/_conf.py -> neo4j-5.2.dev0/neo4j
copying neo4j/_data.py -> neo4j-5.2.dev0/neo4j
copying neo4j/_deadline.py -> neo4j-5.2.dev0/neo4j
copying neo4j/_exceptions.py -> neo4j-5.2.dev0/neo4j
copying neo4j/_meta.py -> neo4j-5.2.dev0/neo4j
copying neo4j/_routing.py -> neo4j-5.2.dev0/neo4j
copying neo4j/addressing.py -> neo4j-5.2.dev0/neo4j
copying neo4j/api.py -> neo4j-5.2.dev0/neo4j
copying neo4j/conf.py -> neo4j-5.2.dev0/neo4j
copying neo4j/data.py -> neo4j-5.2.dev0/neo4j
copying neo4j/debug.py -> neo4j-5.2.dev0/neo4j
copying neo4j/exceptions.py -> neo4j-5.2.dev0/neo4j
copying neo4j/meta.py -> neo4j-5.2.dev0/neo4j
copying neo4j/packstream.py -> neo4j-5.2.dev0/neo4j
copying neo4j/routing.py -> neo4j-5.2.dev0/neo4j
copying neo4j.egg-info/PKG-INFO -> neo4j-5.2.dev0/neo4j.egg-info
copying neo4j.egg-info/SOURCES.txt -> neo4j-5.2.dev0/neo4j.egg-info
copying neo4j.egg-info/dependency_links.txt -> neo4j-5.2.dev0/neo4j.egg-info
copying neo4j.egg-info/requires.txt -> neo4j-5.2.dev0/neo4j.egg-info
copying neo4j.egg-info/top_level.txt -> neo4j-5.2.dev0/neo4j.egg-info
copying neo4j/_async/__init__.py -> neo4j-5.2.dev0/neo4j/_async
copying neo4j/_async/bookmark_manager.py -> neo4j-5.2.dev0/neo4j/_async
copying neo4j/_async/driver.py -> neo4j-5.2.dev0/neo4j/_async
copying neo4j/_async/io/__init__.py -> neo4j-5.2.dev0/neo4j/_async/io
copying neo4j/_async/io/_bolt.py -> neo4j-5.2.dev0/neo4j/_async/io
copying neo4j/_async/io/_bolt3.py -> neo4j-5.2.dev0/neo4j/_async/io
copying neo4j/_async/io/_bolt4.py -> neo4j-5.2.dev0/neo4j/_async/io
copying neo4j/_async/io/_bolt5.py -> neo4j-5.2.dev0/neo4j/_async/io
copying neo4j/_async/io/_common.py -> neo4j-5.2.dev0/neo4j/_async/io
copying neo4j/_async/io/_pool.py -> neo4j-5.2.dev0/neo4j/_async/io
copying neo4j/_async/work/__init__.py -> neo4j-5.2.dev0/neo4j/_async/work
copying neo4j/_async/work/result.py -> neo4j-5.2.dev0/neo4j/_async/work
copying neo4j/_async/work/session.py -> neo4j-5.2.dev0/neo4j/_async/work
copying neo4j/_async/work/transaction.py -> neo4j-5.2.dev0/neo4j/_async/work
copying neo4j/_async/work/workspace.py -> neo4j-5.2.dev0/neo4j/_async/work
copying neo4j/_async_compat/__init__.py -> neo4j-5.2.dev0/neo4j/_async_compat
copying neo4j/_async_compat/concurrency.py -> neo4j-5.2.dev0/neo4j/_async_compat
copying neo4j/_async_compat/util.py -> neo4j-5.2.dev0/neo4j/_async_compat
copying neo4j/_async_compat/network/__init__.py -> neo4j-5.2.dev0/neo4j/_async_compat/network
copying neo4j/_async_compat/network/_bolt_socket.py -> neo4j-5.2.dev0/neo4j/_async_compat/network
copying neo4j/_async_compat/network/_util.py -> neo4j-5.2.dev0/neo4j/_async_compat/network
copying neo4j/_async_compat/shims/__init__.py -> neo4j-5.2.dev0/neo4j/_async_compat/shims
copying neo4j/_codec/__init__.py -> neo4j-5.2.dev0/neo4j/_codec
copying neo4j/_codec/hydration/__init__.py -> neo4j-5.2.dev0/neo4j/_codec/hydration
copying neo4j/_codec/hydration/_common.py -> neo4j-5.2.dev0/neo4j/_codec/hydration
copying neo4j/_codec/hydration/_interface/__init__.py -> neo4j-5.2.dev0/neo4j/_codec/hydration/_interface
copying neo4j/_codec/hydration/v1/__init__.py -> neo4j-5.2.dev0/neo4j/_codec/hydration/v1
copying neo4j/_codec/hydration/v1/hydration_handler.py -> neo4j-5.2.dev0/neo4j/_codec/hydration/v1
copying neo4j/_codec/hydration/v1/spatial.py -> neo4j-5.2.dev0/neo4j/_codec/hydration/v1
copying neo4j/_codec/hydration/v1/temporal.py -> neo4j-5.2.dev0/neo4j/_codec/hydration/v1
copying neo4j/_codec/hydration/v2/__init__.py -> neo4j-5.2.dev0/neo4j/_codec/hydration/v2
copying neo4j/_codec/hydration/v2/hydration_handler.py -> neo4j-5.2.dev0/neo4j/_codec/hydration/v2
copying neo4j/_codec/hydration/v2/temporal.py -> neo4j-5.2.dev0/neo4j/_codec/hydration/v2
copying neo4j/_codec/packstream/__init__.py -> neo4j-5.2.dev0/neo4j/_codec/packstream
copying neo4j/_codec/packstream/_common.py -> neo4j-5.2.dev0/neo4j/_codec/packstream
copying neo4j/_codec/packstream/v1/__init__.py -> neo4j-5.2.dev0/neo4j/_codec/packstream/v1
copying neo4j/_spatial/__init__.py -> neo4j-5.2.dev0/neo4j/_spatial
copying neo4j/_sync/__init__.py -> neo4j-5.2.dev0/neo4j/_sync
copying neo4j/_sync/bookmark_manager.py -> neo4j-5.2.dev0/neo4j/_sync
copying neo4j/_sync/driver.py -> neo4j-5.2.dev0/neo4j/_sync
copying neo4j/_sync/io/__init__.py -> neo4j-5.2.dev0/neo4j/_sync/io
copying neo4j/_sync/io/_bolt.py -> neo4j-5.2.dev0/neo4j/_sync/io
copying neo4j/_sync/io/_bolt3.py -> neo4j-5.2.dev0/neo4j/_sync/io
copying neo4j/_sync/io/_bolt4.py -> neo4j-5.2.dev0/neo4j/_sync/io
copying neo4j/_sync/io/_bolt5.py -> neo4j-5.2.dev0/neo4j/_sync/io
copying neo4j/_sync/io/_common.py -> neo4j-5.2.dev0/neo4j/_sync/io
copying neo4j/_sync/io/_pool.py -> neo4j-5.2.dev0/neo4j/_sync/io
copying neo4j/_sync/work/__init__.py -> neo4j-5.2.dev0/neo4j/_sync/work
copying neo4j/_sync/work/result.py -> neo4j-5.2.dev0/neo4j/_sync/work
copying neo4j/_sync/work/session.py -> neo4j-5.2.dev0/neo4j/_sync/work
copying neo4j/_sync/work/transaction.py -> neo4j-5.2.dev0/neo4j/_sync/work
copying neo4j/_sync/work/workspace.py -> neo4j-5.2.dev0/neo4j/_sync/work
copying neo4j/graph/__init__.py -> neo4j-5.2.dev0/neo4j/graph
copying neo4j/spatial/__init__.py -> neo4j-5.2.dev0/neo4j/spatial
copying neo4j/time/__init__.py -> neo4j-5.2.dev0/neo4j/time
copying neo4j/time/__main__.py -> neo4j-5.2.dev0/neo4j/time
copying neo4j/time/_arithmetic.py -> neo4j-5.2.dev0/neo4j/time
copying neo4j/time/_clock_implementations.py -> neo4j-5.2.dev0/neo4j/time
copying neo4j/time/_metaclasses.py -> neo4j-5.2.dev0/neo4j/time
copying neo4j/time/arithmetic.py -> neo4j-5.2.dev0/neo4j/time
copying neo4j/time/clock_implementations.py -> neo4j-5.2.dev0/neo4j/time
copying neo4j/time/hydration.py -> neo4j-5.2.dev0/neo4j/time
copying neo4j/time/metaclasses.py -> neo4j-5.2.dev0/neo4j/time
copying neo4j/work/__init__.py -> neo4j-5.2.dev0/neo4j/work
copying neo4j/work/query.py -> neo4j-5.2.dev0/neo4j/work
copying neo4j/work/summary.py -> neo4j-5.2.dev0/neo4j/work
copying testkitbackend/__init__.py -> neo4j-5.2.dev0/testkitbackend
copying testkitbackend/__main__.py -> neo4j-5.2.dev0/testkitbackend
copying testkitbackend/_driver_logger.py -> neo4j-5.2.dev0/testkitbackend
copying testkitbackend/_warning_check.py -> neo4j-5.2.dev0/testkitbackend
copying testkitbackend/backend.py -> neo4j-5.2.dev0/testkitbackend
copying testkitbackend/exceptions.py -> neo4j-5.2.dev0/testkitbackend
copying testkitbackend/fromtestkit.py -> neo4j-5.2.dev0/testkitbackend
copying testkitbackend/server.py -> neo4j-5.2.dev0/testkitbackend
copying testkitbackend/test_subtest_skips.py -> neo4j-5.2.dev0/testkitbackend
copying testkitbackend/totestkit.py -> neo4j-5.2.dev0/testkitbackend
copying testkitbackend/_async/__init__.py -> neo4j-5.2.dev0/testkitbackend/_async
copying testkitbackend/_async/backend.py -> neo4j-5.2.dev0/testkitbackend/_async
copying testkitbackend/_async/requests.py -> neo4j-5.2.dev0/testkitbackend/_async
copying testkitbackend/_sync/__init__.py -> neo4j-5.2.dev0/testkitbackend/_sync
copying testkitbackend/_sync/backend.py -> neo4j-5.2.dev0/testkitbackend/_sync
copying testkitbackend/_sync/requests.py -> neo4j-5.2.dev0/testkitbackend/_sync
Writing neo4j-5.2.dev0/setup.cfg
Creating tar archive
removing 'neo4j-5.2.dev0' (and everything under it)
```

Which indicates that this change doesn't do anything except for with it, setuptools complains about `no previously-included directories found matching 'tests'`. Did it make more of a difference for you?

Remarks:
 * The testkitbackend is bundled, wich is definitely some dev code that shouldn't be shipped. So thanks already for pointing me towards that.
 * To do the release we actually don't run `python setup.py sdist` but `./bin/make-dist 5.2.0` or what ever version we want to release.

In fact, I realized that setuptools recursively filters out the `tests` package even without listing it explicitly. I then dug around the docs and found [this](https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#flat-layout) which confirms my observation.

I suggest changing the exclusion to `test*`. That would (redundantly) match `tests` as well as `testkitbackend` and other test-ish packages that might be added in the future as well as all their sub-packages.

```suggestion
packages = find_packages(exclude=["test*"])
```

Comment (robsdedude) [reply setup.py]: I'd like to get this into 5.3 which will be cut off in about a week. @thesamesam What do you think about my comment?

**Review by robsdedude — APPROVED** [review]
Since I haven't heard back, I went ahead and merged my suggestions so that this change can make it into 5.3

:monkey_face:

**Pull request merged**